src/geometry/adjoints.jl

```diff
@@ -0,0 +1,58 @@
+function ChainRulesCore.rrule(::Interpolations.BSplineInterpolation, T, A, it, axs)
+    y = Interpolations.BSplineInterpolation(T, A, it, axs)
+    function bspline_pb(Δy)
+        return NoTangent(), NoTangent(), Δy, NoTangent(), NoTangent()
+    end
+    return y, bspline_pb
+end
+
+function ChainRulesCore.rrule(::Type{Interpolations.BSplineInterpolation{T, N, TA, IT, TAX}}, A, axs, it) where {T, N, TA, IT, TAX}
+    y = Interpolations.BSplineInterpolation{T, N, TA, IT, TAX}(A, axs, it)
+    function bspline_const_pb(Δy)
+        return NoTangent(), Δy, NoTangent(), NoTangent()
+    end
+    return y, bspline_const_pb
+end
+
+function ChainRulesCore.rrule(::Type{Interpolations.Extrapolation{T, N, ITPT, IT, ET}}, itp, et::Union{Interpolations.Flat, Interpolations.Reflect, Interpolations.Line}) where {T,N,ITPT,IT,ET}
+    y = Interpolations.Extrapolation{T,N,ITPT,IT,ET}(itp, et)
+    function extrap_const_pb(Δy)
+        return NoTangent(), Δy, NoTangent()
+    end
+    return y, extrap_const_pb
+end
+
+Zygote.@nograd Interpolations.tweight
+Zygote.@nograd Interpolations.tcoef
+
+function ChainRulesCore.rrule(::typeof(Interpolations.copy_with_padding), T, A, it)
```

Comment:
Does this need an rrule or can it be a nograd?

Reply:
I think we can do a nograd, but it will collide with the fact that we plan to use the image gradients ∂L/∂X at a later part of our project.

Reply:
Yeah, just check what the impact is and we can take a call then